### ISSUE TITLE
Look out for v7 adjusted branches on subdataset save/add

### ISSUE THREAD

**Issue by mih** [datalad/datalad#3178]
At the moment we are not considering adjusted branches at all, and treat them like any other branch. I fear that this is not OK. Here is a demo:

```
(datalad3-dev) mih@meiner /tmp/demo % dl rev-create .
[INFO   ] Creating a new annex repo at /tmp/demo 
create(ok): /tmp/demo (dataset)                                                              
(datalad3-dev) mih@meiner /tmp/demo (git)-[master] % DATALAD_REPO_VERSION=7 dl rev-create sub
[INFO   ] Creating a new annex repo at /tmp/demo/sub 
create(ok): /tmp/demo/sub (dataset)
(datalad3-dev) 1 mih@meiner /tmp/demo (git)-[master] % git -C sub annex adjust --unlock
adjust 
Switched to branch 'adjusted/master(unlocked)'
ok
(datalad3-dev) mih@meiner /tmp/demo (git)-[master] % git -C sub status
On branch adjusted/master(unlocked)
nothing to commit, working tree clean
(datalad3-dev) mih@meiner /tmp/demo (git)-[master] % dl rev-save
save(ok): . (dataset)                                                                        
(datalad3-dev) mih@meiner /tmp/demo (git)-[master] % git diff HEAD~1 |cat
diff --git a/.gitmodules b/.gitmodules
new file mode 100644
index 0000000..c489803
--- /dev/null
+++ b/.gitmodules
@@ -0,0 +1,3 @@
+[submodule "sub"]
+       path = sub
+       url = ./sub
diff --git a/sub b/sub
new file mode 160000
index 0000000..9ed87e3
--- /dev/null
+++ b/sub
@@ -0,0 +1 @@
+Subproject commit 9ed87e39e73f3b80d17fea655ff880eb7f77fa5f
(datalad3-dev) mih@meiner /tmp/demo (git)-[master] % git -C sub log |cat
commit 9ed87e39e73f3b80d17fea655ff880eb7f77fa5f
Author: Michael Hanke <michael.hanke@gmail.com>
Date:   Mon Feb 25 11:13:15 2019 +0100

    git-annex adjusted branch

commit faccd4df8b3b6a6c8e8a25aa4c676cd99e327f55
Author: Michael Hanke <michael.hanke@gmail.com>
Date:   Mon Feb 25 11:13:15 2019 +0100

    [DATALAD] new dataset
```
It is the tip of the adjusted branch that is recorded at the subdataset state.

I believe that the correct thing to do (re a later publish), would be to detect that this is an adjusted branch of `master`, perform a `git annex sync` (or `git annex --unlock`) to bring the actual `master` branch up to speed wrt to the state of the adjusted branch, and then register the state of `master` in the parent.

However, this will result in a permanent `modified` status report for the subdataset (despite an empty diff), as it would need to go back to a checkout of the adjusted branch for any actual use.

Not sure at all how to deal with this.

Thoughts?

**Comment by kyleam** [comment]
That is quite problematic.  I'm not seeing an obvious way to deal with it either.

**Comment by mih** [comment]
FTR: nothing has changed in this regard.

**Comment by mih** [comment]
Addressed in #4275 and tracked in #3818
